File: Assignment-2/Question-1/q1.py
def is_equal(l1, l2, n, m):
    if n != m:
        return False

    count_map = {}
    for i in range(n):
        count_map[l1[i]] = count_map.get(l1[i], 0) + 1

    for i in range(n):
        if l2[i] not in count_map or count_map[l2[i]] == 0:
            return False
        count_map[l2[i]] -= 1

    return True

if __name__ == "__main__":
    l1 = [1, 2, 3, 4, 5, 6]
    l2 = [3, 6, 2, 4, 1, 5]
    n = len(l1)
    m = len(l2)

    if is_equal(l1, l2, n, m):
        print("l1 and l2 are equal")
    else:
        print("l1 and l2 are not equal")

File: Assignment-2/Question-1/q1.cpp
#include <iostream>
#include <unordered_map>
using namespace std;

bool isEqual(int l1[], int l2[], int n, int m){
    if(n!=m){
        return false;
    }

    unordered_map<int,int> map;
    for(int i -0;i<n;i++)
        map[l1[i]]++;

    for(int i = 0;i<n;i++){
        if(map.find(l2[i]) == map.end())
            return false;
        if(map[l2[i]] == 0)
            return false;
        map[l2[i]]--;
    }

    return true;
}

int main(){
    int l1[] = {1,2,3,4,5,6};
    int l2[] = {3,6,2,4,1,5};
    int n = l1.size();
    int m = l2.size();
    if(isEqual(l1,l2,n,m))
        cout << "l1 and l2 are equal";
    else
        cout << "l1 and l2 are not equal";
    return 0;
}
